core/src/main/java/xyz/gianlu/librespot/player/feeders/storage/ChannelManager.java

```diff
@@ -160,16 +166,24 @@ void streamError(short code) {
 
             @Override
             public void run() {
+                LOGGER.trace("ChannelManager.Handler is starting");
+
                 while (true) {
+                    // TODO Check if loop breaks at Session.close()
```

Comment:
The problem here is that there is no logic to even close the Channel when it's not being used anymore, but we could interrupt everything when the ChannelManager, I'll leave it up to you because I don't know it should be done properly from the ExecutorService.

Reply:
I will look into this, thank you.

Reply:
Okay, it seems like the ChannelManager.Handler-Thread is finishing instantly in normal situations. So I would just ignore this ;)